fenv_lmi.cpp

```diff
@@ -59,24 +60,19 @@ namespace floating_point_environment {} // doxygen workaround.
 
 void fenv_initialize()
 {
-#if defined LMI_IEC_559
+#if defined LMI_X87
+    x87_control_word(default_x87_control_word());
+#else  // !defined LMI_X87
     fenv_t save_env;
     feholdexcept(&save_env);
     fesetround(FE_TONEAREST);
-#   if defined __MINGW32__
-    // Here is an example of a C99 7.6/9 extension that controls
-    // hardware precision.
-    fesetenv(FE_PC64_ENV);
-#   else  // !defined __MINGW32__
-#       error Find a platform-specific way to set hardware precision.
-#   endif // !defined __MINGW32__
-#elif defined LMI_X86
-    x87_control_word(default_x87_control_word());
-#else  // Unknown compiler or platform.
-#   error Unknown compiler or platform.
-#endif // Unknown compiler or platform.
+
+    // We currently don't set hardware precision in this case as this is an
+    // x87-specific feature.
+#endif // !defined LMI_X87
 }
 
+#if defined LMI_X87
 e_ieee754_precision fenv_precision()
 {
 #if defined __BORLANDC__
```

Comment:
Dropping support for Borland could simplify the preprocessor checks in this file a bit, and I think it should be done, but I decided against doing it in this patch to keep the changes well separate. Please let me know if you think I should do this right now.